plugins/env/app/assets/javascripts/env/application.js

```diff
@@ -158,4 +158,12 @@
     });
   });
 
+  // Disable environment inputs if "use config service" checked
+  $(document).ready(function() {
+    var envVarInputs = $('.environment-variable-inputs');
+    $('#use_config_service').change(function() {
+      envVarInputs.prop('disabled', this.checked);
```

Comment:
should only disable them if they are empty ... otherwise the existing ones would still stay ... 
prefer allowing mixing for an easy migration